core/src/main/scala/org/broadinstitute/dsde/rawls/bucketMigration/BucketMigrationService.scala

```diff
@@ -21,6 +29,73 @@ class BucketMigrationService(val dataSource: SlickDataSource, val samDAO: SamDAO
     with WorkspaceSupport
     with LazyLogging {
 
+  def getBucketMigrationProgressForWorkspace(
+    workspaceName: WorkspaceName
+  ): Future[Option[MultiregionalBucketMigrationProgress]] =
+    asFCAdmin {
+      for {
+        workspace <- getV2WorkspaceContext(workspaceName)
+        res <- dataSource.inTransaction(getBucketMigrationProgress(workspace))
+      } yield res
+    }
+
+  def getBucketMigrationProgressForBillingProject(
+    billingProjectName: RawlsBillingProjectName
+  ): Future[Map[String, Option[MultiregionalBucketMigrationProgress]]] =
+    asFCAdmin {
+      for {
+        workspaces <- dataSource.inTransaction { dataAccess =>
+          dataAccess.workspaceQuery.listWithBillingProject(billingProjectName)
+        }
+        progress <- workspaces.traverse { workspace =>
+          dataSource
+            .inTransaction(getBucketMigrationProgress(workspace))
+            .recover(_ => None)
```

Comment:
Don't want to throw a 404 if a single workspace hasn't been migrated yet, so catch the 404 from `getBucketMigrationProgress` and return None instead